File: src/infrastructure/di/RoleService.cs
using core.Account;
using core.fs.Adapters.Authentication;

namespace di;

public class RoleService(string? adminEmail) : IRoleService
{
    public bool IsAdmin(UserState user) => !string.IsNullOrWhiteSpace(adminEmail) && user.Email == adminEmail;

    public string? GetAdminEmail() => adminEmail;
}

File: src/infrastructure/di.fs/RoleService.fs
namespace di

open core.Account
open core.fs.Adapters.Authentication

type RoleService(adminEmail: string option) =
    
    interface IRoleService with
        member _.IsAdmin(user: UserState) = 
            match adminEmail with
            | Some email when not (System.String.IsNullOrWhiteSpace(email)) -> 
                user.Email = email
            | _ -> false
        
        member _.GetAdminEmail() = 
            match adminEmail with
            | Some email -> email
            | None -> null
